main/src/mpu.cpp

```diff
@@ -0,0 +1,157 @@
+#include "mpu.h"
+#include "MPU6050.h"
+
+namespace mpu {
+
+MPU6050 mpudev;
+
+static SemaphoreHandle_t mutex = xSemaphoreCreateMutex();
+static unsigned int count = 0;
+static Data internal{};
+static float gyro_res = 0, accel_res = 0;
+static bool init_success = false;
+
+/**
+ * @brief Mpu initialization. Return gracefully on initialisation error.
+ */
+void init() {
+    // Parameters change the resolution of the sensor.
+    mpudev.initialize(/*MPU6050_ACCEL_FS_2, MPU6050_GYRO_FS_250*/);
+
+    // Calibration on start?
+    // mpudev.CalibrateAccel(10);
+    // mpudev.CalibrateGyro(10);
+
+    // Alternatively, we can set offsets obtained by running IMU_Zero
+    // example from the library.
+    // mpudev.setXAccelOffset(0);
+    // mpudev.setYAccelOffset(0);
+    // mpudev.setZAccelOffset(0);
+    // mpudev.setXGyroOffset(0);
+    // mpudev.setYGyroOffset(0);
+    // mpudev.setZGyroOffset(0);
+
+    if (!mpudev.testConnection()) {
+        // die
+        Serial.println("Connection to mpu failed");
+        return;
+    }
+
+    if (mutex == nullptr) {
+        // die
+        Serial.println("Couldn't allocate resources for mutex!");
+        return;
+    }
+
+    init_success = true;
+    accel_res = mpudev.get_acce_resolution() / 2;
+    gyro_res = mpudev.get_gyro_resolution() / 2;
+}
+
+/**
+ * @brief Consume and return collected data.
+ *        ! CAN BLOCK INDEFINITELY (if INCLUDE_vTaskSuspend == 1) !
+ * @return Max g force, Average g force, Average rotation.
+ */
+Data get_mpu() {
+    xSemaphoreTake(mutex, portMAX_DELAY);
+
+    Data dane{
+        .max_x = internal.max_x,
+        .max_y = internal.max_y,
+        .max_z = internal.max_z,
+        .avg_x = internal.avg_x / count, // Instead of counting could implement
+        .avg_y = internal.avg_y / count, // moving average, but dunno if it's
+        .avg_z = internal.avg_z / count, // superior and/or viable alternative.

```

Comment:
Don't write comments in this format, either use /* */ or write it in one line (and remove this comment)

Reply:
Removed comment, although in my opinion it's worth to discuss using moving average.